clang/test/SemaSYCL/array-kernel-param.cpp

```diff
@@ -165,10 +167,12 @@ int main() {
 // Check Kernel_StructAccArray parameters
 // CHECK: FunctionDecl {{.*}}Kernel_StructAccArray{{.*}} 'void (__global int *, sycl::range<1>, sycl::range<1>, sycl::id<1>, __global int *, sycl::range<1>, sycl::range<1>, sycl::id<1>)'
 // CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc '__global int *'
+// CHECK-NEXT: SYCLAccessorPtrAttr
```

Comment:
Can you add ESIMD test to show that SYCLAccessorPtrAttr gets generated now? Thanks

Reply:
I'm surprised we don't have existing AST tests for this :/

Reply:
@elizabethandrews
@smanna12 
ESIMD support is provided by the explicit SIMD SYCL extension, where the lambda function passed to the kernel invocation functions (eg.parallel_for) is marked with a special attribute - `SYCL_ESIMD_KERNEL`. `(__attribute__((sycl_explicit_simd))`
I don't see how Abstract Syntax Tree can be generated/applicable for an extension, which is probably why 
there are no AST related tests for ESIMD kernels.

I have instead included a test case for ESIMD kernel which captures an accessor and made sure the `!kernel_arg_accessor_ptr` which confirms that `SYCLAccessorPtrAttr` was created for ESIMD kernels as well.